File: Arrays/anagram.py
# 242. Valid Anagram

class Solution:
    def isAnagram(self, s, t):
        if len(s) != len(t):
            return False
        
        sCharMap = [0] * 26
        tCharMap = [0] * 26
        for i in range(0, len(s)):
            a = ord('a')
            sCharMap[ord(s[i]) - a] += 1
            tCharMap[ord(t[i]) - a] += 1
        
        print(sCharMap)
        print(tCharMap)
        return sCharMap == tCharMap
    

print(Solution().isAnagram('rat', 'car'))
        

File: Arrays/anagram.java
import java.util.Arrays;

class Solution {
    public boolean isAnagram(String s, String t) {
        if(s.length() != t.length()) {
            return false;
        }
        int[] sCharMap = new int[26];
        int[] tCharMap = new int[26];
        char offsetChar = 'a';
        
        for (int i = 0; i < s.length(); i++) {
            sCharMap[(int)s.charAt(i) - (int)offsetChar] += 1;
            tCharMap[(int)t.charAt(i) - (int)offsetChar] += 1;
        }
        
        return Arrays.equals(sCharMap, tCharMap);
    }
}